File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> mp = new HashMap<>();
        int n = nums.length, sum = 0, c = 0;
        mp.put(0, 1);

        for (int i = 0; i < n; i++) {
            sum += nums[i];
            if (mp.containsKey(sum - k))
                c += mp.get(sum - k);
            mp.put(sum, mp.getOrDefault(sum, 0) + 1);
        }
        return c;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        hmap={0:1}
        prefix=0
        res=0
        for num in nums:
            prefix+=num
            diff=prefix-k
            if diff in hmap:
                res+=hmap[diff]
            hmap[prefix]=1+hmap.get(prefix,0)
        print(hmap)        
        return res            
                